packages/vite/src/node/plugins/resolve.ts

```diff
@@ -621,6 +632,30 @@ export function tryNodeResolve(
   })!
 
   if (!pkg) {
+    // if import can't be found, check if it's an optional peer dep.
+    // if so, we can resolve to a special id that errors only when imported.
+    if (
+      basedir !== root && // root has no peer dep
+      !isBuiltin(id) &&
```

Comment:
These `id` references should be using `nestedPath` instead

Reply:
@aleclarson I worry we will end up unable to keep track of these comments if you only add them to merged PRs. Would it be possible for you to create a new issue with a reproduction linking to them? Or a PR with a failing test case that others can later work on? If we don't have an open issue or PR in a few days these will be completely buried by new activity.

Reply:
I made the fix at https://github.com/vitejs/vite/pull/10593